pkg/services/ngalert/api/compat.go

```diff
@@ -342,3 +343,33 @@ func NilIfEmpty[T any](v *[]T) *[]T {
 	}
 	return v
 }
+
+func AlertingFileExportFromMuteTimings(orgID int64, m []definitions.MuteTimeInterval) definitions.AlertingFileExport {
+	f := definitions.AlertingFileExport{
+		APIVersion:  1,
+		MuteTimings: make([]definitions.MuteTimeIntervalExport, 0, len(m)),
+	}
+	for _, mi := range m {
+		f.MuteTimings = append(f.MuteTimings, MuteTimeIntervalExportFromMuteTiming(orgID, mi))
+	}
+	return f
+}
+
+func MuteTimeIntervalExportFromMuteTiming(orgID int64, m definitions.MuteTimeInterval) definitions.MuteTimeIntervalExport {
+	return definitions.MuteTimeIntervalExport{
+		OrgID:            orgID,
+		MuteTimeInterval: m.MuteTimeInterval,
+	}
+}
+
+// Converts definitions.MuteTimeIntervalExport to definitions.MuteTimeIntervalExportHcl using JSON marshalling. Returns error if structure could not be marshalled\unmarshalled
+func MuteTimingIntervalToMuteTimeIntervalHclExport(m definitions.MuteTimeIntervalExport) (definitions.MuteTimeIntervalExportHcl, error) {
+	result := definitions.MuteTimeIntervalExportHcl{}
+	j := jsoniter.ConfigCompatibleWithStandardLibrary
+	mdata, err := j.Marshal(m)
+	if err != nil {
+		return result, err
+	}
+	err = j.Unmarshal(mdata, &result)
+	return result, err
```

Comment:
This feels potentially dangerous since the type being marshaled from is in Prometheus, but is not in the prometheus `config` package. Prometheus could change this type, and if that happened the compiler wouldn't flag this when doing the upgrade. I realize directly mapping the fields is a lot of boilerplate, but it seems quite a bit safer, and also less expensive.

Reply:
It looks like @julienduchesne has done some heavy-lifting and decoupled API models from Prometheus' https://github.com/grafana/grafana/pull/79339. 
Also, if the changes in that PR do not land, changes in upstream models will break unit-tests I added in this PR.